meta-luneos/conf/distro/include/luneos-preferred-providers.inc

```diff
@@ -1,10 +1,40 @@
-# Copyright (c) 2012-2014 LG Electronics, Inc.
+# Copyright (c) 2012-2023 LG Electronics, Inc.
 
 # Place distro-wide PREFERRED_PROVIDER settings here; place MACHINE-specific
 # settings in <MACHINE>.conf .
 
-PREFERRED_PROVIDER_virtual/webruntime = "webruntime"
-VIRTUAL-RUNTIME_webruntime = "webruntime"
+# Empty will use default gcc or you can use -clang"
+WEBRUNTIME_TOOLCHAIN = ""
+
+SKIP_RECIPE[jsoncpp-clang] ?= "${@oe.utils.conditional('WEBRUNTIME_TOOLCHAIN', '-clang', '', 'WEBRUNTIME_TOOLCHAIN is not set to -clang', d)}"
+SKIP_RECIPE[libpbnjson-clang] ?= "${@oe.utils.conditional('WEBRUNTIME_TOOLCHAIN', '-clang', '', 'WEBRUNTIME_TOOLCHAIN is not set to -clang', d)}"
+SKIP_RECIPE[media-resource-calculator-clang] ?= "${@oe.utils.conditional('WEBRUNTIME_TOOLCHAIN', '-clang', '', 'WEBRUNTIME_TOOLCHAIN is not set to -clang', d)}"
+SKIP_RECIPE[umediaserver-clang] ?= "${@oe.utils.conditional('WEBRUNTIME_TOOLCHAIN', '-clang', '', 'WEBRUNTIME_TOOLCHAIN is not set to -clang', d)}"
+SKIP_RECIPE[wam-clang] ?= "${@oe.utils.conditional('WEBRUNTIME_TOOLCHAIN', '-clang', '', 'WEBRUNTIME_TOOLCHAIN is not set to -clang', d)}"
+SKIP_RECIPE[googletest-clang] ?= "${@oe.utils.conditional('WEBRUNTIME_TOOLCHAIN', '-clang', '', 'WEBRUNTIME_TOOLCHAIN is not set to -clang', d)}"
+SKIP_RECIPE[media-codec-interface-clang] ?= "${@oe.utils.conditional('WEBRUNTIME_TOOLCHAIN', '-clang', '', 'WEBRUNTIME_TOOLCHAIN is not set to -clang', d)}"
+SKIP_RECIPE[g-media-pipeline-clang] ?= "${@oe.utils.conditional('WEBRUNTIME_TOOLCHAIN', '-clang', '', 'WEBRUNTIME_TOOLCHAIN is not set to -clang', d)}"
+
+PREFERRED_PROVIDER_llvm = "${@oe.utils.conditional('WEBRUNTIME_TOOLCHAIN', '-clang', 'clang', 'llvm', d)}"
+PREFERRED_PROVIDER_llvm-native = "${@oe.utils.conditional('WEBRUNTIME_TOOLCHAIN', '-clang', 'clang-native', 'llvm-native', d)}"
+PREFERRED_PROVIDER_nativesdk-llvm = "${@oe.utils.conditional('WEBRUNTIME_TOOLCHAIN', '-clang', 'nativesdk-clang', 'nativesdk-llvm', d)}"
+PROVIDES:pn-clang = "llvm"
+PROVIDES:pn-clang-native = "llvm-native"
+PROVIDES:pn-nativesdk-clang = "nativesdk-llvm"
+
+PREFERRED_PROVIDER_virtual/webruntime = "webruntime${WEBRUNTIME_TOOLCHAIN}"
+VIRTUAL-RUNTIME_webruntime = "webruntime${WEBRUNTIME_TOOLCHAIN}"
+
+#FixMe: LuneOS db8 doesn't like below somehow.
```

Comment:
I'm getting this:
"ERROR: db8-3.2.0-28+gitAUTOINC+3962b2d169-r40 do_package_qa: QA Issue: /usr/lib/db8/bin/errorOpenTempDb.bash contained in package db8 requires /bin/bash, but no providers found in RDEPENDS:db8? [file-rdeps]
ERROR: db8-3.2.0-28+gitAUTOINC+3962b2d169-r40 do_package_qa: QA Issue: /usr/lib/db8/tests/volume/runtest.bash contained in package db8-tests requires /bin/bash, but no providers found in RDEPENDS:db8-tests? [file-rdeps]
ERROR: db8-3.2.0-28+gitAUTOINC+3962b2d169-r40 do_package_qa: Fatal QA errors were found, failing task.
ERROR: Logfile of failure stored in: /media/herrie/LuneOS/kirkstone/webos-ports/tmp-glibc/work/core2-64-webos-linux/db8/3.2.0-28+gitAUTOINC+3962b2d169-r40/temp/log.do_package_qa.1813147
ERROR: Task (/media/herrie/LuneOS/kirkstone/webos-ports/meta-webos-ports/meta-luneos/recipes-webos/db8/db8.bb:do_package_qa) failed with exit code '1'
"

Reply:
Is this with VIRTUAL-RUNTIME_bash set to bash?

Reply:
Yes, in the DB8 recipe: VIRTUAL-RUNTIME_bash ?= "bash"

Reply:
That's weak assignment in db8 recipe, but doesn't matter if you uncomment the 3 lines here.

But you would still need to use bash as bash provider not busybox, because unlike OSE we don't enable ASH_IS_BASH in busybox defconfig.

Reply:
meta-webos/recipes-core/busybox/busybox/defconfig
...
CONFIG_BASH_IS_ASH=y
...